# Midtest Python-Case dataset : Titanic Dataset

This case is used Titanic dataset.

The dataset contains data for 887 of the real Titanic passengers. 

Each row represents one person. The columns describe different attributes about the person including whether they survived (S), their age (A), their passenger-class (C), their sex (G) and the fare they paid (X). The table below shows the Data Dictionary.

## Import Libraries

In [21]:
import pandas as pd
import numpy as np

## Import Dataset

In [22]:
#Dataset is already located in one file directory

data = pd.read_csv('titanic.csv')
data.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

### 1. Get Dimension (col,row) of dataframe

In [23]:
data.shape

(887, 8)

Dataset of Titanic above have 887 rows and 8 columns.

### 2. Get Information of data type for each variable

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
Survived                   887 non-null int64
Pclass                     887 non-null int64
Name                       887 non-null object
Sex                        887 non-null object
Age                        887 non-null float64
Siblings/Spouses Aboard    887 non-null int64
Parents/Children Aboard    887 non-null int64
Fare                       887 non-null float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.5+ KB


From the function (.info()) above will result data type each variable of dataset Titanic. Variables that have an integer(int) data type are Survived, Pclass, Siblings/Spouses Aboard, then variables that have float data type are Age, Fare and variables that have object data type are Name, Sex. 

### 3. Get Infromation of how many passengers survived and not survived

In [25]:
pd.value_counts(data['Survived'].values, sort=False)

0    545
1    342
dtype: int64

The above Syntax calculates the amount of survived(1) and not survived(0) from 'Survived' columns. From this, it appears that the number of survived(1) passenger is 545, and the number of passenger who is not survived(0) is 342. Therefore, the number of passengers who survived less than the number of unsaved passengers.

### 4. Drop columns 'Name'

In [26]:
data = data.drop('Name', axis = 1)

### 5. Add one new column 'Family'

In [27]:
data['family'] = data['Siblings/Spouses Aboard']+data['Parents/Children Aboard']
data.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  family  
0                        0   7.2500       1  
1                        0  71.2833       1  
2                        0   7.9250       0  
3                        0  53.1000       1  
4                        0   8.0500       0

Add column 'Family' that is represented number family-member aboard. That formula is sum of 'Siblings/Spouses Aboard' and 'Parents/Children Aboard'.

### 6. Add new column 'Age_miss'

### 6.1 Add new column 'Age_miss'

In [28]:
# Create New column based on conditional statement 

data['Age_miss']=np.where(data['Age'].notnull,'No', 'Yes')
data.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  family Age_miss  
0                        0   7.2500       1       No  
1                        0  71.2833       1       No  
2                        0   7.9250       0       No  
3                        0  53.1000       1       No  
4                        0   8.0500       0       No

### 7. Fill Missing Values 'Age' with mean of 'Age'

### 7.1 Checking Missing values of dataset Titanic

In [29]:
#Checking Missing Values

data.isnull().sum()

Survived                   0
Pclass                     0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
family                     0
Age_miss                   0
dtype: int64

After checking missing values, this data is doesn't have missing values.

### 7.2 Fill Missing values with Mean

The syntax below can be used if there are missing values in the 'Age'. This syntax is replaced 'missing values' with mean of this column 'Age'.

data['Age].fillna(data['Age'].mean(), inplace=True)

### 8. Maximum Passenger Age who survived 

In [30]:
#Get new dataframe which summarizes 'survived' passenger datasets
datasurvived=data[data['Survived']==1]

#Get maximum 'Age' of survived(1) passenger
datasurvived['Age'].max()

80.0

Maximum Age of passenger who survived is 80 years old.

### 9. Get Information passesnger survived from each 'Pclass'

In [31]:
#Filter data, select only passengers who Survived
Survivedlife= data['Survived']==1

#Count of passengers who survived that group by 'Pclass'
Survivedlife.groupby(data['Pclass']).count()
groupsurvived.count()

Pclass
1    216
2    184
3    487
Name: Survived, dtype: int64

At first glance, it looked like mayoritas pasengers who survived from 3rd class, the second one is 1st class and the last is 2nd class. 

### 10. Split data into titanic1 and titanic2 with proportion 70% and 30%

In [32]:
#Split dataset into titanic1 and titanic2 randomly

#Using this packages to split data 
from sklearn.model_selection import train_test_split

#Split data into two, which proportion of test or titanic2 is 30% of data
titanic1 ,titanic2 = train_test_split(data,test_size=0.3)  

In [33]:
#Dimension of titanic1 data

titanic1.shape

(620, 9)

Its mean titanic1 data have 620 rows and 9 columns.

In [34]:
#Looking data titanic1 first 10 rows

titanic1.head(10)

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
579         0       2    male  54.0                        0   
571         0       3    male  16.0                        0   
810         0       3    male  30.5                        0   
269         1       3    male  25.0                        0   
848         0       3  female   9.0                        1   
287         1       3  female  22.0                        0   
518         0       3    male  22.0                        0   
723         1       3  female  28.0                        0   
107         0       3    male  38.0                        0   
401         0       3    male  28.0                        1   

     Parents/Children Aboard     Fare  family Age_miss  
579                        0  26.0000       0       No  
571                        0   8.0500       0       No  
810                        0   8.0500       0       No  
269                        0   0.0000       0       No  
848                        1  15.2458       2       No  
287                        0   7.7500       0       No  
518                        0   7.8958       0       No  
723                        0   7.7375       0       No  
107                        0   7.8958       0       No  
401                        0  15.8500       1       No

In [35]:
#Dimension of titanic2 dataset

titanic2.shape

(267, 9)

Its mean titanic2 data have 267 rows and 9 columns.

In [36]:
#Looking titanic2 dataset first 10 rows

titanic2.head(10)

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
718         0       2    male  34.0                        0   
119         0       2    male  21.0                        2   
566         1       3    male  32.0                        0   
826         1       3  female  15.0                        1   
54          1       1    male  46.0                        0   
746         1       2  female   4.0                        1   
217         1       1  female  32.0                        0   
824         1       3    male  19.0                        0   
69          0       2    male  32.0                        0   
622         0       1    male  61.0                        0   

     Parents/Children Aboard     Fare  family Age_miss  
718                        0  13.0000       0       No  
119                        0  73.5000       2       No  
566                        0   7.8542       0       No  
826                        0  14.4542       1       No  
54                         0  35.5000       0       No  
746                        1  23.0000       2       No  
217                        0  76.2917       0       No  
824                        0   7.7500       0       No  
69                         0  10.5000       0       No  
622                        0  32.3208       0       No